From: Russ Allbery <rra@cpan.org>
Date: Sat, 11 Jun 2022 16:33:06 -0700
Subject: Fix t/spin/errors with Perl 5.36

The Text::Balanced in Perl 5.36 uses a different default regex for
extracting text, which changes the error output.  Adjust for that
in the test suite.
---
 t/spin/errors.t | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/t/spin/errors.t b/t/spin/errors.t
index c35669b..235240e 100755
--- a/t/spin/errors.t
+++ b/t/spin/errors.t
@@ -19,7 +19,7 @@ use Test::More tests => 2;
 # not entirely correct because line tracking is very complicated and still not
 # entirely correct.
 my $EXPECTED_ERRORS = <<'ERRORS';
-errors.th:1: cannot find argument 2: Did not find opening bracket after prefix: "\s*", detected at offset 2
+errors.th:1: cannot find argument 2: Did not find opening bracket after prefix: <PREFIX>, detected at offset 2
 errors.th:3: invalid macro placeholder \2 (greater than 1)
 errors.th:5: invalid macro argument count for \badcount
 errors.th:9: unknown variable \=UNKNOWN
@@ -40,7 +40,9 @@ my ($stdout, $stderr) = capture {
     $spin->spin_thread_file($input);
 };
 
-# Simplify the file name, and then check against the expected output.
+# Simplify the file name and search prefix, and then check against the
+# expected output.
 $stderr =~ s{ ^ [^:]+/errors[.]th: }{errors.th:}xmsg;
 $stderr =~ s{ (cannot [ ] stat [ ] file [ ]) /[^:]+/([^/:]+) : .* }{$1$2\n}xms;
+$stderr =~ s{ (prefix: [ ]) \"[^\"]+\", }{$1<PREFIX>,}xms;
 is($stderr, $EXPECTED_ERRORS, 'errors are correct');
